src/Microsoft.DocAsCode.Build.Engine/TemplatePreprocessors/TemplateUtility.cs

```diff
@@ -38,5 +38,20 @@ public string ResolveSourceRelativePath(string originPath, string currentFileOut
                 return originPath;
             }
         }
+
+        public string UpdateHref(string originalHref, string sourceFileKey)
```

Comment:
It's different. DocumentBuildContext.UpdateHrefCore converts href starting with `~/` to the final href, while this PR's function converts original href to href starting with `~/`.